tests/test_task_worker.py

```diff
@@ -930,38 +923,3 @@ def slow_task():
 
         assert updated_task_run_1.state.is_completed()
         assert updated_task_run_2.state.is_scheduled()
-
```

Comment:
`serve` directly passes through to `TaskWorker` and we have more or less the same exact test via `test_task_worker_respects_limit`

Reply:
Could you update this test to ensure that `serve` passes its `limit` parameter through to the `TaskWorker`? That'll help make sure the correct plumbing stays in place.

Reply:
yep can do! i still have a weird flake to figure out here - some worker thread dying somehow